lib/awful/tag.lua

```diff
@@ -749,6 +766,32 @@ function tag.object.set_layout(t, layout)
     return layout
 end
 
+function tag.object.get_layouts(self)
+    -- The override has the priority.
+    local cls = tag.getproperty(self, "custom_layouts")
+
+    -- Allow custom layouts to be listed until GC-ed.
+    if not cls then
+        cls = setmetatable({}, {__mode="v"})
+        tag.setproperty(self, "custom_layouts", cls)
+    end
+
+    -- Required to get the default/fallback list of layouts
+    alayout = alayout or require("awful.layout")
+
+    -- Add the custom and unlisted layouts to te front
+    if (not gtable.hasitem(cls, self.layout))
+      and not alayout.get_tag_layout_index(self) then
+        table.insert(cls, self.layout)
+    end
+
+    return gtable.merge(gtable.clone(cls), alayout.layouts)
+end
+
+function tag.object.set_layouts(self, layouts)
+    tag.setproperty(self, "layouts", layouts)
```

Comment:
Where is this property `get`ed? Or does setting a property directly mean that the getter is no longer used?

Reply:
refactoring facepalm